## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :x: [more...](#0cc7ca26-1f0a-4406-b77c-776f9244c757) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#8a767501-5673-434f-a36b-42e2e7118b58) |
| mint-pools.sh | :x: [more...](#20330cd6-bf61-43ce-a102-773fe8d52a00) |
<a name="8a767501-5673-434f-a36b-42e2e7118b58"></a>
### 13830-32d80f3/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s6:32715
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-mcg5h0442miehg",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 365,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-k8s6:32715/aws-sdk-go-test-mcg5h0442miehg\": dial tcp 15.15.15.6:32715: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... FAILED in 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-mcg5h0442miehg",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 365,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-k8s6:32715/aws-sdk-go-test-mcg5h0442miehg\": dial tcp 15.15.15.6:32715: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-php tests ... FAILED in 6 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-mcg5h0442miehg",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 365,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-k8s6:32715/aws-sdk-go-test-mcg5h0442miehg\": dial tcp 15.15.15.6:32715: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-ruby tests ... FAILED in 8 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 6146.22,
  "error": "Failed to open TCP connection to minio-k8s6:32715 (Connection refused - connect(2) for \"minio-k8s6\" port 32715)",
  "status": "FAIL"
}
(1/15) Running awscli tests ... FAILED in 9 seconds
{"name": "awscli", "duration": 7786, "function": "make_bucket\n", "status": "FAIL", "error": "Could not connect to the endpoint URL: "http://minio-k8s6:32715/awscli-mint-test-bucket-23320""}
(1/15) Running healthcheck tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": null,
  "duration": 1,
  "error": "Get \"http://minio-k8s6:32715/minio/health/live\": dial tcp 15.15.15.6:32715: connect: connection refused",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/15) Running mc tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": null,
  "duration": 1,
  "error": "Get \"http://minio-k8s6:32715/minio/health/live\": dial tcp 15.15.15.6:32715: connect: connection refused",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/15) Running minio-dotnet tests ... done in 44 seconds
(2/15) Running minio-go tests ... FAILED in 10 seconds
{
  "args": {
    "bucketName": "minio-go-test-yhuxvkf3bstth6lj",
    "region": "eu-west-1"
  },
  "duration": 10229,
  "error": "Put \"http://minio-k8s6:32715/minio-go-test-yhuxvkf3bstth6lj/\": dial tcp 15.15.15.6:32715: connect: connection refused",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testMakeBucketErrorV2",
  "status": "FAIL"
}
(2/15) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 54,
  "status": "FAIL",
  "error": "java.net.ConnectException: Failed to connect to minio-k8s6/15.15.15.6:32715 >>> [okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:297), okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207), okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226), okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106), okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74), okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall.execute(RealCall.kt:154), io.minio.S3Base.execute(S3Base.java:507), io.minio.MinioClient.makeBucket(MinioClient.java:1192), FunctionalTest.testBucketApi(FunctionalTest.java:447), FunctionalTest.testBucketApiCases(FunctionalTest.java:475), FunctionalTest.makeBucket(FunctionalTest.java:506), FunctionalTest.runBucketTests(FunctionalTest.java:3644), FunctionalTest.runTests(FunctionalTest.java:3718), FunctionalTest.runEndpointTests(FunctionalTest.java:3818), FunctionalTest.main(FunctionalTest.java:3938)]"
}
(2/15) Running minio-js tests ... FAILED in 2 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 1034,
  "status": "FAIL",
  "error": "Error: connect ECONNREFUSED 15.15.15.6:32715 at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1159:16)"
}
(2/15) Running minio-py tests ... FAILED in 9 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-ba456e5d-3827-4c4c-a0a8-e1ca51de51d0",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='minio-k8s6', port=32715): Max retries exceeded with url: /minio-py-test-ba456e5d-3827-4c4c-a0a8-e1ca51de51d0 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f0540069340>: Failed to establish a new connection: [Errno 111] Connection refused'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 174, in _new_conn\n    conn = connection.create_connection(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 96, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 86, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 699, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 394, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 239, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.8/http/client.py\", line 1252, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1298, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1247, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1007, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.8/http/client.py\", line 947, in send\n    self.connect()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 205, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 186, in _new_conn\n    raise NewConnectionError(\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f0540069340>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 168, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 612, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 783, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 783, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 783, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 755, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 574, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s6', port=32715): Max retries exceeded with url: /minio-py-test-ba456e5d-3827-4c4c-a0a8-e1ca51de51d0 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f0540069340>: Failed to establish a new connection: [Errno 111] Connection refused'))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 9050
}
(2/15) Running s3cmd tests ... FAILED in 1 seconds
{
  "name": "s3cmd",
  "duration": "1213",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions-(FAQ)\n  If the error persists, please report the\n  following lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\nInvoked as: /usr/local/bin/s3cmd --config /tmp/.s3cmd-13610/s3cfg mb s3://s3cmd-test-bucket-25533\nProblem: <class 'ConnectionRefusedError: [Errno 111] Connection refused\nS3cmd:   2.2.0\npython:   3.8.10 (default, Sep 28 2021, 16:10:42) \n[GCC 9.3.0]\nenvironment LANG=C.UTF-8\n\nTraceback (most recent call last):\n  File \"/usr/local/bin/s3cmd\", line 3209, in <module>\n    rc = main()\n  File \"/usr/local/bin/s3cmd\", line 3106, in main\n    rc = cmd_func(args)\n  File \"/usr/local/bin/s3cmd\", line 260, in cmd_bucket_create\n    response = s3.bucket_create(uri.bucket(), cfg.bucket_location, cfg.extra_headers)\n  File \"/usr/local/lib/python3.8/dist-packages/S3/S3.py\", line 430, in bucket_create\n    response = self.send_request(request)\n  File \"/usr/local/lib/python3.8/dist-packages/S3/S3.py\", line 1480, in send_request\n    conn = ConnMan.get(self.get_hostname(resource['bucket']))\n  File \"/usr/local/lib/python3.8/dist-packages/S3/ConnMan.py\", line 284, in get\n    conn.c.connect()\n  File \"/usr/lib/python3.8/http/client.py\", line 918, in connect\n    self.sock = self._create_connection(\n  File \"/usr/lib/python3.8/socket.py\", line 808, in create_connection\n    raise err\n  File \"/usr/lib/python3.8/socket.py\", line 796, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions-(FAQ)\n  If the error persists, please report the\n  above lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
}
(2/15) Running s3select tests ... FAILED in 9 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-c9578015-60a3-4ed8-b68c-bdb1c2353762"
  },
  "duration": 8069,
  "message": "HTTPConnectionPool(host='minio-k8s6', port=32715): Max retries exceeded with url: /s3select-test-c9578015-60a3-4ed8-b68c-bdb1c2353762 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f40d0d52130>: Failed to establish a new connection: [Errno 111] Connection refused'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 174, in _new_conn\n    conn = connection.create_connection(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 96, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 86, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 699, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 394, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 239, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.8/http/client.py\", line 1252, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1298, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1247, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1007, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.8/http/client.py\", line 947, in send\n    self.connect()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 205, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 186, in _new_conn\n    raise NewConnectionError(\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f40d0d52130>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 612, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 783, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 783, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 783, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 755, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 574, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s6', port=32715): Max retries exceeded with url: /s3select-test-c9578015-60a3-4ed8-b68c-bdb1c2353762 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f40d0d52130>: Failed to establish a new connection: [Errno 111] Connection refused'))\n",
  "status": "FAIL"
}
(2/15) Running versioning tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-vstuzfdgpw51gr"
  },
  "duration": 1315,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-k8s6:32715/versioning-test-vstuzfdgpw51gr\": dial tcp 15.15.15.6:32715: connect: connection refused",
  "function": "/mint/build/versioning/bucket.go:47: main.testMakeBucket",
  "message": "Versioning CreateBucket Failed",
  "name": "versioning",
  "status": "FAIL"
}

Executed 1 out of 15 tests successfully.
```
<a name="20330cd6-bf61-43ce-a102-773fe8d52a00"></a>
### 13830-32d80f3/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s7:31094
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 4 minutes and 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-g1k1oe4ih0hn51",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 242443,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-k8s7:31094/aws-sdk-go-test-g1k1oe4ih0hn51\": dial tcp 15.15.15.7:31094: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... FAILED in 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-g1k1oe4ih0hn51",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 242443,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-k8s7:31094/aws-sdk-go-test-g1k1oe4ih0hn51\": dial tcp 15.15.15.7:31094: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-php tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-g1k1oe4ih0hn51",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 242443,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-k8s7:31094/aws-sdk-go-test-g1k1oe4ih0hn51\": dial tcp 15.15.15.7:31094: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-ruby tests ... FAILED in 4 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 2111.78,
  "error": "Failed to open TCP connection to minio-k8s7:31094 (Connection refused - connect(2) for \"minio-k8s7\" port 31094)",
  "status": "FAIL"
}
(1/15) Running awscli tests ... FAILED in 13 seconds
{
  "name": "awscli",
  "duration": 11834,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}
(1/15) Running healthcheck tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": null,
  "duration": 1,
  "error": "Get \"http://minio-k8s7:31094/minio/health/live\": dial tcp 15.15.15.7:31094: connect: connection refused",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/15) Running mc tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": null,
  "duration": 1,
  "error": "Get \"http://minio-k8s7:31094/minio/health/live\": dial tcp 15.15.15.7:31094: connect: connection refused",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/15) Running minio-dotnet tests ... done in 3 seconds
(2/15) Running minio-go tests ... FAILED in 4 seconds
{
  "args": {
    "bucketName": "minio-go-test-kpq3yq6aqjpgo0el",
    "region": "eu-west-1"
  },
  "duration": 4275,
  "error": "Server not initialized, please try again.",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testMakeBucketErrorV2",
  "status": "FAIL"
}
(2/15) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 58,
  "status": "FAIL",
  "error": "java.net.ConnectException: Failed to connect to minio-k8s7/15.15.15.7:31094 >>> [okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:297), okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207), okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226), okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106), okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74), okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall.execute(RealCall.kt:154), io.minio.S3Base.execute(S3Base.java:507), io.minio.MinioClient.makeBucket(MinioClient.java:1192), FunctionalTest.testBucketApi(FunctionalTest.java:447), FunctionalTest.testBucketApiCases(FunctionalTest.java:475), FunctionalTest.makeBucket(FunctionalTest.java:506), FunctionalTest.runBucketTests(FunctionalTest.java:3644), FunctionalTest.runTests(FunctionalTest.java:3718), FunctionalTest.runEndpointTests(FunctionalTest.java:3818), FunctionalTest.main(FunctionalTest.java:3938)]"
}
(2/15) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 4,
  "status": "FAIL",
  "error": "Error: connect ECONNREFUSED 15.15.15.7:31094 at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1159:16)"
}
(2/15) Running minio-py tests ... FAILED in 4 minutes and 1 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-7720b0cc-a516-49a9-a4e6-a504a72206d9",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='minio-k8s7', port=31094): Max retries exceeded with url: /minio-py-test-7720b0cc-a516-49a9-a4e6-a504a72206d9 (Caused by ResponseError('too many 503 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 168, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 612, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 783, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 783, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 783, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 846, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 846, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 574, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s7', port=31094): Max retries exceeded with url: /minio-py-test-7720b0cc-a516-49a9-a4e6-a504a72206d9 (Caused by ResponseError('too many 503 error responses'))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 241362
}
(2/15) Running s3cmd tests ... FAILED in 0 seconds
{
  "name": "s3cmd",
  "duration": "241",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions-(FAQ)\n  If the error persists, please report the\n  following lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\nInvoked as: /usr/local/bin/s3cmd --config /tmp/.s3cmd-31683/s3cfg mb s3://s3cmd-test-bucket-12262\nProblem: <class 'ConnectionRefusedError: [Errno 111] Connection refused\nS3cmd:   2.2.0\npython:   3.8.10 (default, Sep 28 2021, 16:10:42) \n[GCC 9.3.0]\nenvironment LANG=C.UTF-8\n\nTraceback (most recent call last):\n  File \"/usr/local/bin/s3cmd\", line 3209, in <module>\n    rc = main()\n  File \"/usr/local/bin/s3cmd\", line 3106, in main\n    rc = cmd_func(args)\n  File \"/usr/local/bin/s3cmd\", line 260, in cmd_bucket_create\n    response = s3.bucket_create(uri.bucket(), cfg.bucket_location, cfg.extra_headers)\n  File \"/usr/local/lib/python3.8/dist-packages/S3/S3.py\", line 430, in bucket_create\n    response = self.send_request(request)\n  File \"/usr/local/lib/python3.8/dist-packages/S3/S3.py\", line 1480, in send_request\n    conn = ConnMan.get(self.get_hostname(resource['bucket']))\n  File \"/usr/local/lib/python3.8/dist-packages/S3/ConnMan.py\", line 284, in get\n    conn.c.connect()\n  File \"/usr/lib/python3.8/http/client.py\", line 918, in connect\n    self.sock = self._create_connection(\n  File \"/usr/lib/python3.8/socket.py\", line 808, in create_connection\n    raise err\n  File \"/usr/lib/python3.8/socket.py\", line 796, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions-(FAQ)\n  If the error persists, please report the\n  above lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
}
(2/15) Running s3select tests ... FAILED in 6 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-0458ceb4-d270-4bdd-a10b-713320b84989"
  },
  "duration": 6019,
  "message": "HTTPConnectionPool(host='minio-k8s7', port=31094): Max retries exceeded with url: /s3select-test-0458ceb4-d270-4bdd-a10b-713320b84989 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f3bb34aa100>: Failed to establish a new connection: [Errno 111] Connection refused'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 174, in _new_conn\n    conn = connection.create_connection(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 96, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 86, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 699, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 394, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 239, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.8/http/client.py\", line 1252, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1298, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1247, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1007, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.8/http/client.py\", line 947, in send\n    self.connect()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 205, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 186, in _new_conn\n    raise NewConnectionError(\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f3bb34aa100>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 612, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 783, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 783, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 783, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 755, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 574, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-k8s7', port=31094): Max retries exceeded with url: /s3select-test-0458ceb4-d270-4bdd-a10b-713320b84989 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f3bb34aa100>: Failed to establish a new connection: [Errno 111] Connection refused'))\n",
  "status": "FAIL"
}
(2/15) Running versioning tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-lu4l94tw6axdsn"
  },
  "duration": 395,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-k8s7:31094/versioning-test-lu4l94tw6axdsn\": dial tcp 15.15.15.7:31094: connect: connection refused",
  "function": "/mint/build/versioning/bucket.go:47: main.testMakeBucket",
  "message": "Versioning CreateBucket Failed",
  "name": "versioning",
  "status": "FAIL"
}

Executed 1 out of 15 tests successfully.
```
<a name="0cc7ca26-1f0a-4406-b77c-776f9244c757"></a>
### 13830-32d80f3/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s1:32134
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 1 minutes and 50 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 28 seconds
(8/15) Running minio-dotnet tests ... done in 50 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 8 seconds
(10/15) Running minio-java tests ... done in 28 seconds
(11/15) Running minio-js tests ... FAILED in 57 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook for \"removeObject(bucketName, objectName, removeOpts)",
  "args": "bucketName:minio-js-test-ver-presign55a5ce3f-5167-4afa-a6d5-78da8ec93254 ",
  "duration": 5,
  "alert": "datafile-1-b ",
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty. You must delete all versions in the bucket. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:89:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:164:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:88:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1334:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(11/15) Running minio-py tests ... done in 1 minutes and 12 seconds
(12/15) Running s3cmd tests ... done in 25 seconds
(13/15) Running s3select tests ... done in 3 seconds
(14/15) Running versioning tests ... done in 3 minutes and 4 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/bucket-handlers.go
+++ b/cmd/bucket-handlers.go
@@ -1210,7 +1210,7 @@ func (api objectAPIHandlers) HeadBucketHandler(w http.ResponseWriter, r *http.Re
 		return
 	}
 
-	writeSuccessResponseHeadersOnly(w)
+	writeResponse(w, http.StatusOK, nil, mimeXML)
 }
 
 // DeleteBucketHandler - Delete bucket

--- a/cmd/erasure-server-pool.go
+++ b/cmd/erasure-server-pool.go
@@ -1131,7 +1131,9 @@ func (z *erasureServerPools) ListObjects(ctx context.Context, bucket, prefix, ma
 	}
 	merged, err := z.listPath(ctx, &opts)
 	if err != nil && err != io.EOF {
-		logger.LogIf(ctx, err)
+		if !isErrBucketNotFound(err) {
+			logger.LogIf(ctx, err)
+		}
 		return loi, err
 	}
 

--- a/cmd/metacache-entries.go
+++ b/cmd/metacache-entries.go
@@ -307,7 +307,7 @@ func (m metaCacheEntries) resolve(r *metadataResolutionParams) (selected *metaCa
 		// shallow decode.
 		xl, err := entry.xlmeta()
 		if err != nil {
-			logger.LogIf(context.Background(), err)
+			logger.LogIf(GlobalContext, err)
 			continue
 		}
 		objsValid++
@@ -339,9 +339,15 @@ func (m metaCacheEntries) resolve(r *metadataResolutionParams) (selected *metaCa
 	if objsValid < r.objQuorum {
 		return nil, false
 	}
+
 	// If all objects agree.
 	if selected != nil && objsAgree == objsValid {
 		return selected, true
+	}
+
+	// If cached is nil we shall skip the entry.
+	if selected.cached == nil {
+		return nil, false
 	}
 
 	// Merge if we have disagreement.

